### DISCUSSION TITLE
How to extract only the body text of the PDF, not the bulleted points, headings and subheadings

### DISCUSSION THREAD

**Discussion by Kituva** [jsvine/pdfplumber#707] [Ask for help with specific PDFs]
I need to extract text from a PDF. I extracted the same using extract_text() function. I tried to split the text into sentences using tokenize. The problem here is it split the lines and headings into one single sentence. I do not require the headings and subheadings in the PDF. How to overcome this problem?

Below is the PDF file
[sample.pdf](https://github.com/jsvine/pdfplumber/files/9290648/sample.pdf)

**Comment by jsvine** [comment]
Hi @Kituva, and thanks for your interest in this library. I'm not sure I 100% understand the question. Could you provide sample code you're using, the output it produces, and how that differs from what you wanted?

If I do understand the question, it is how to extract the only the body text of the PDF, not the headings and subheading. For that, I'd suggest using the following approaches, after having examined your specific PDF:

```python
page = pdf.pages[0]
without_headings = page.filter(lambda obj: "Bold" not in obj.get("fontname", ""))
print(without_headings.extract_text())
```

**Reply by Kituva** [reply]
```
import pdfplumber
import re

ecdata = ""

with pdfplumber.open("XYZ Transcript.pdf") as pdf:
    for i in range(len(pdf.pages)):
        print("Pgae No.: ", i+1)
        page_obj = pdf.pages[i]
        page = page_obj.within_bbox((70, 50, page_obj.width, 790))
        ecpagedata = page.extract_text()
        ecdata += ecpagedata
        print(page.extract_text())
```
**This is the output of the above code**

![image](https://user-images.githubusercontent.com/110910255/184076662-0ccc15d7-a6cd-4331-81c4-a4a549e98b9d.png)

**What I need as output is only the complete sentences of the file and not the unwanted bullets, heading and subheadings**

Good day, and thank you for standing by. Welcome to the XYZ Second Quarter 2099 
Earnings Conference Call. At this time, all participants are in a listen-only mode. After the 
speakers' presentation, there will be a question-and-answer session. (Operator 
Instructions) Please be advised that today's conference is being recorded.  
 
I would now like to hand the conference over to your speaker today, Alpha, Vice President 
of Investor Relations. Please go ahead.  

Thank you, operator. Good afternoon and welcome to XYZ’s second quarter 2022 
earnings call. I'm joined today by Bravo, XYZ’s Founder and CEO; and Charlie, our CFO. 
Full details of our results and additional management commentary are available in our 
shareholder letter, which can be found on our Investor Relations website at 
website.com/investor. Our comments and responses to your questions on this call reflect 
management's views as of today only and we disclaim any obligation to update this 
information. On this call, we'll make forward-looking statements which are predictions, 
projections, or other.

**This is the source file**

_The source file is my own creation and does not directly or indirectly represent any entity real or fictitious whatsoever._
[XYZ Transcript.pdf](https://github.com/jsvine/pdfplumber/files/9306693/XYZ.Transcript.pdf)

Kindly help me out to get the required output. @jsvine

**Reply by jsvine** [reply]
Hi @Kituva, and thank you for the additional details. PDF documents don't have an explicit concept of a heading or bullet points, but rather simply represent those through different fonts / fontsizes, indentation, et cetera. If you have a document with predictable heading and bullet-point characteristics, you could write code that specifically filters those out. But if you have a range of document layouts / formatting styles, you might instead want to come up with more-general heuristics.

**Reply by Kituva** [reply]
Hi @jsvine, Thanks for the help.
